Inline assembly fail to build on non-x86 architectures

Hi.  When using this header library with coz-profiler in Debian, https://tracker.debian.org/pkg/coz-profiler , we discovered that the library contain inline assembly that only work on x86 architectures.  Can the code be made more portable?  This is the build error on arm64:

```
/tmp/ccjpqVMY.s: Assembler messages:
/tmp/ccjpqVMY.s:2491: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:2700: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:11219: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:11255: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:11543: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:11728: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:13510: Error: unknown mnemonic `pause' -- `pause'
/tmp/ccjpqVMY.s:13538: Error: unknown mnemonic `pause' -- `pause'
../common.mk:67: recipe for target 'obj/profiler.o' failed
```

--- a/spinlock.h
+++ b/spinlock.h
@@ -7,7 +7,15 @@ class spinlock {
 public:
   inline void lock() {
     while(_flag.test_and_set()) {
-      __asm__ __volatile__ ("pause");
+#if defined(__i386__) || defined(__x86_64__)
+      /*
+       * NOTE: "rep nop" works on all Intel architectures and has the same
+       * encoding as "pause" on the newer ones.
+       */
+      __asm__ __volatile__ ("rep nop");
+#else
+      /* nothing */
+#endif
     }
   }
   

--- a/spinlock.h
+++ b/spinlock.h
@@ -7,7 +7,7 @@ class spinlock {
 public:
   inline void lock() {
     while(_flag.test_and_set()) {
-      __asm__("pause");
+      __asm__ __volatile__ ("pause");
     }
   }
   

